include/deal.II/base/function_spherical.h

```diff
@@ -0,0 +1,122 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__function_spherical_h
+#define dealii__function_spherical_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/std_cxx11/array.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Functions
+{
+
+  /**
+   * An abstract base class for a scalar-valued function $f=f(r,\theta,\phi)$
+   * defined in spherical coordinates. This class wraps transformation of values,
+   * gradients and hessians from spherical coordinates to Cartesian coordinate system.
+   * Therefore derived classes only need to implement those functions in
+   * spherical coordinates (specifically svalue(), sgradient() and
+   * shessian() ). The convention for angles is the same as in
+   * GeometricUtilities::Coordinates.
+   *
+   * @note This function is currently only implemented for dim==3 .
+   *
+   * @author Denis Davydov, 2017
+   */
+  template <int dim>
+  class Spherical : public Function<dim>
+  {
+  public:
+    /**
+     * Constructor which should be provided with @p center defining the origin
+     * of the coordinate system.
+     */
+    Spherical(const Point<dim> &center);
+
+    /**
+     * Virtual destructor.
+     */
+    virtual ~Spherical();
+
+    /**
+     * Return the value of the function at the given point.
+     */
+    virtual double value (const Point<dim> &point,
```

Comment:
maybe say "This function converts the given point to spherical coordinates, calls svalue() with it, and converts the result into Cartesian coordinates."

Same for the other two functions.